Gems/Atom/Feature/Common/Code/Include/Atom/Feature/CoreLights/LightCommon.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Aabb.h>
+#include <AzCore/Math/Frustum.h>
+#include <AzCore/Math/Hemisphere.h>
+#include <AzCore/Math/Sphere.h>
+#include <AzCore/std/containers/variant.h>
+
+#include <Atom/Feature/Mesh/MeshCommon.h>
+
+namespace AZ::RPI
+{
+    class Scene;
+}
+
+namespace AZ::Render::LightCommon
+{
+    using LightBounds = AZStd::variant<Sphere, Hemisphere, Frustum, Aabb>;
+
+    template <typename BoundsType>
+    struct EmptyFilter
+    {
+        constexpr bool operator()(const BoundsType&) const { return true; }
+    };
+
+    template <typename BoundsType, class Filter = EmptyFilter<BoundsType>>
+    void MarkMeshesWithLightType(AZ::RPI::Scene* scene, AZStd::span<const BoundsType> bounds, AZ::RPI::Cullable::FlagType flag, Filter filter = {})
```

Comment:
I originally had everything in `LightCommon.h` and moved out the `MarkMeshesForBounds()` to `MeshCommon.h` since it seemed like a generic piece. But I suppose I could consider `LightBounds` to just be any bounds type that is supported by `ShapeIntersection`, and move it all out to `MeshCommon.h`